middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -107,7 +107,12 @@ let block_load ~dbg (kind : P.Block_access_kind.t) (mutability : Mutability.t)
   | Mixed { field_kind = Flat_suffix field_kind; _ } -> (
     match field_kind with
     | Imm -> C.get_field_computed Immediate mutability ~block ~index dbg
-    | Float | Float64 -> C.unboxed_float_array_ref block index dbg)
+    | Float | Float64 ->
+      (* CR layouts v5.1: We should use the mutability here. *)
```

Comment:
Maybe clarify the comment a bit: the current code behaves as if this read is always mutable, but when it is immutable we could potentially generate different/better code, if I understand correctly.

Reply:
Good idea, done.